### PR TITLE
[NDMII-2934]tag interface metrics with internal resource

### PR DESCRIPTION
**Pull request by jedupau**
### What does this PR do?

add an internal tag to the snmp interface metrics that will make those metrics be enriched by the tags present in the corresponding REDAPL resource

### Motivation

We want to start tagging our interfaces with tags stored in REDAPL

### CODE REVIEW

**Review by FlorianVeaux — DISMISSED** [review]
I think that it might be dangerous to extract the device_id from the tags. It's not clear that this is a dependency to this function and may start silently failing.

Maybe we should have the device_id as a param to `reportColumnMetrics`, or as a field of `MetricSender`?


Also, let's make sure to unit test this new behavior

File: pkg/collector/corechecks/snmp/internal/report/report_metrics.go [hunk 1]
```diff
@@ -153,6 +160,8 @@ func (ms *MetricSender) reportColumnMetrics(metricConfig profiledefinition.Metri
 						log.Tracef("unable to tag snmp.%s metric with interface_config data: %s", symbol.Name, err.Error())
 					}
 					tmpTags = append(tmpTags, interfaceCfg.Tags...)
+
+					tmpTags = append(tmpTags, fmt.Sprintf("dd.internal.resource:ndm_interface_user_tags:%s:%s", deviceID, fullIndex))
```
Comment (FlorianVeaux) [inline pkg/collector/corechecks/snmp/internal/report/report_metrics.go:164]: **💬 suggestion:** ‏Minor suggestion, we could benefit from having a util method for that. Especially as we'll use the same thing for devices

**Comment by FlorianVeaux** [comment]
Also I think the no-changelog label is a mistake?

File: pkg/collector/corechecks/snmp/internal/report/report_metrics.go (hunk 1 above)
Comment (jedupau) [reply pkg/collector/corechecks/snmp/internal/report/report_metrics.go]: https://github.com/DataDog/datadog-agent/pull/27628/commits/ab05a64e749d46dd57b952a50727775860b8ac2c

**Comment by jedupau** [comment]
@FlorianVeaux I added a commit which adds the device_id as a param of `ReportMetrics` and adds a changelog

File: releasenotes/notes/user-tag-interface-metrics-597f442a7de2787b.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+enhancements:
+  - |
+    Add a new tag to interface metrics that will allow them to be tagged by user tags.
```
Comment (drichards-87) [inline releasenotes/notes/user-tag-interface-metrics-597f442a7de2787b.yaml:11]:
```suggestion
    Add a new tag to interface metrics that will allow them to be tagged by user-defined tags.
```

**Review by drichards-87 — APPROVED** [review]
Left a small suggestion from Docs and approved the PR.

File: releasenotes/notes/user-tag-interface-metrics-597f442a7de2787b.yaml (hunk 2 above)
Comment (drichards-87) [reply releasenotes/notes/user-tag-interface-metrics-597f442a7de2787b.yaml:11]: As an alternative, I would also suggest considering something along the lines of: 

"Add the ability to tag interface metrics with user-defined tags."

You mention that the PR adds an internal tag, so maybe users don't need to know that specific detail.

**Review by jmw51798 — APPROVED** [review]

**Review by NouhaManai96 — APPROVED** [review]

**Review by TCheruy — APPROVED** [review]

File: releasenotes/notes/user-tag-interface-metrics-597f442a7de2787b.yaml (hunk 2 above)
Comment (jedupau) [reply releasenotes/notes/user-tag-interface-metrics-597f442a7de2787b.yaml]: updated eb6f55bc31088f140c2f47aebe8750bc5d99140b

**Pull request merged**